### PR TITLE
[POC] Add Turborepo 

### PR DESCRIPTION
**Pull request by aacebo**
This is a POC on how using Vercels https://turbo.build/repo package might make the management of this monorepo more streamlined. 

Turborepo is a tool used to manage a monorepos build/test/lint pipelines and shared dependencies. Usage looks like:

`yarn install`
`npx turbo build`
`npx turbo lint`
`npx turbo clean`

I've also added some changes that would allow us to upgrade packages to their latest versions and unify dependency versions across the samples/lib

### CODE REVIEW

File: js/packages/teams-ai/src/AI.ts [hunk 1]
```diff
@@ -18,7 +18,6 @@ import { TurnState } from './TurnState';
 
 /**
  * A function that can be used to select a prompt to use for the current turn.
- *
  * @template TState Type of the turn state.
```
Comment (corinagum) [inline js/packages/teams-ai/src/AI.ts:21]: :o jsdocs should make up its mind. we just added these :/

**Comment by corinagum** [comment]
@aacebo could you add a summary proposal as the top item in this thread? :)

File: js/packages/teams-ai/src/AI.ts (hunk 1 above)
Comment (aacebo) [reply js/packages/teams-ai/src/AI.ts:21]: ya I think this is because I was able to update alot of the package versions during this process 🤷‍♂️

Comment (corinagum) [reply js/packages/teams-ai/src/AI.ts:21]: 🙃 Resolving - just me complaining

**Pull request closed without merging**